Add page to view all projects of a user / group

valentinleistner opened the pull request:
- adds a page /\<user-or-group-name\> to view all enabled woodpecker repos of a user / group
- also includes a link from the repo page to the owner page
- related to #468

https://user-images.githubusercontent.com/77448526/152249235-95bee6d6-81f4-40f6-8608-77d0f0361476.mov

6543 commented:
CI issue is unrelated ...

anbraten commented on web/src/views/ReposOwner.vue:
```diff
@@ -0,0 +1,63 @@
+    <div class="flex flex-row flex-wrap border-b pb-4 mb-4 items-center dark:border-dark-200">
+      <h1 class="text-xl text-gray-500">{{ repoOwner }}</h1>
+      <TextField v-model="search" class="w-auto md:ml-auto" placeholder="Search ..." />
+      <Button class="md:ml-auto" :to="{ name: 'repo-add' }" start-icon="plus" text="Add repository" />
```
I think we can remove this button from this page. As it could be confusing to a user having a look at some public org / other user and seeing a "I can add something to your org" button.

```suggestion
```

anbraten commented on web/src/views/ReposOwner.vue:
```diff
@@ -0,0 +1,63 @@
+        @click="$router.push({ name: 'repo', params: { repoName: repo.name, repoOwner: repo.owner } })"
+      >
+        <span class="text-gray-500">{{ `${repo.name}` }}</span>
+      </ListItem>
```
```suggestion
      </ListItem>
      <div v-if="searchedRepos.length === 0">
        <span class="text-gray-500 m-auto">This organization / user does not have any projects yet.</span>
      </div>
```

anbraten reviewed (commented):
Nice PR and great work. Just for interest: Do you used our [UI development docs](https://woodpecker-ci.org/docs/development/ui) or did you managed to setup the woodpecker UI development on your own? Seems like you did some frontend dev before 😉 

We (I) need to test if / how this currently works for other users. Will a user easily see all public repos of a different owner?

valentinleistner commented:
> Just for interest: Do you used our [UI development docs](https://woodpecker-ci.org/docs/development/ui) or did you managed to setup the woodpecker UI development on your own?

I first cloned the repo and tried to run the UI from VS Code which worked after I had found out that I needend to install `yarn` by taking a look at `launch.json`. To connect the UI with a server I used the [documentation](https://woodpecker-ci.org/docs/development/ui). Since I was already running Woodpecker in Docker on my server I mostly copied the config to my local machine and added `WOODPECKER_DEV_WWW_PROXY=http://host.docker.internal:8010` (localhost doesn't work because the container has its own IP address) as an environment variable.

> Seems like you did some frontend dev before 😉

I have done some plain HTML, CSS and PHP (using no frameworks and almost no JS (even for things like [tabs](https://css-tricks.com/the-checkbox-hack/))), a bit UIKit, a half finished iOS (& macOS) app using SwiftUI and an almost finished Flutter app. I also have a project which I stared using plain HTML, PHP and JS, added Vue.js, then switched to [Flutter](https://flutter.dev) because at the time I was also working on the other Flutter app and I was planning to eventually build a mobile app anyways. When the Fluter app was almost in a usable state I found that the glitches and huge bundle size were not ideal for a browser first app which recently caused me to completely rewrite the frontend using [React Native](https://reactnative.dev) (using [Expo](https://expo.dev) for crossplatform on web, iOS and Android).

> Will a user easily see all public repos of a different owner?

I am currently using the same `RepoStore` and `useRepoSearch(repos, search)` as for the [existing repos page](), where repos is filtered by the owner. It might be possible to optimize this by only fetching the owner's repos (and not all other repos, too) from the server.

6543 commented:
I'm okay on how it works, but like to refactor the getRepos api ... to have optional filters based on name ... owner ... usw

but that's for another pull :)

6543 commented:
@valentinleistner I only found one nit, before its good to go :)

the space for a user/org repo list between items do not exist:
![Bildschirmfoto zu 2022-02-04 13-49-51](https://user-images.githubusercontent.com/24977596/152532046-ec3dbf90-36b6-4e62-8188-8c8538ae3fd3.png)

should look like:
![Bildschirmfoto zu 2022-02-04 13-49-39](https://user-images.githubusercontent.com/24977596/152532072-e5b74b42-3599-4573-98de-92b0986b7c72.png)

anbraten commented:
The problem is a different here tough. The current implementation of the getRepos endpoint only returns repos / projects you own somehow, but in this case we need to get all public projects someone else owns. We may need to think about adding a new property `owned: true` or similar to the project. This would also help with pipeline runs of public repos appearing in your sidebar as there is a difference in I own the project and I can access the project as it is public or internal.

6543 commented:
well if that's already n issue, we can fix it within this pull ...

6543 commented on web/src/views/ReposOwner.vue:
```diff
@@ -0,0 +1,63 @@
+
+  setup(props) {
+    const repoStore = RepoStore();
+    const repos = computed(() => Object.values(repoStore.repos).filter((v) => v.owner === props.repoOwner));
```
```suggestion
    // TODO: filter server side
    const repos = computed(() => Object.values(repoStore.repos).filter((v) => v.owner === props.repoOwner));
```

6543 commented:
I did add a todo ... as this refactor wont make it into v0.15.0 ... :)
